Adding encoding to import function

netmonk opened the pull request:
After defining a latin1 database, there are error when importing sql scripts encoded in latin1 charset with the import function. 

`mysql` has an option `--default-character-set` to specify the encoding of the input file. This commit modify the import function so that if `encoding` exists when using the import function, it will add `--default-character-set="value of encoding"` to the options of mysql command.  This way encoded file in latin1 for example will be imported without error (by default, the file is considered utf8 encoded)

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
mysql_db supporting import respecting provided `encoding` value. 

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

netmonk committed:
Adding encoding to import function

After defining a latin1 database, there are error when importing sql scripts encoded in latin1 charset with the import function. 

`mysql` has an option `--default-character-set` to specify the encoding of the input file. This commit modify the import function so that if `encoding` exists when using the import function, it will add `--default-character-set="value of encoding"` to the options of mysql command.  This way encoded file in latin1 for example will be imported without error (by default, the file is considered utf8 encoded)

netmonk committed:
mysql_db.py fix 

fixed lint issue, line too long
fixed encoding is not none, neither empty (when there import function is called without a declared encoding value, encoding="" which leads to mysql error for unknown charset)

netmonk committed:
mysql_db.py fix lint 

fixing lint error

netmonk commented:
Anyone reviewing ? @Alexander198961  @Andersson007  @Xyon  @bmalynovytch  @bmildren  
 @kurtdavis  @michaelcoburn @Oneiroi @tolland

bmalynovytch commented:
Hi @netmonk !
Thank you for your contribution.
I'm gonna give a look at it.
Nobody reviewed it as you forgot to add the label saying you were done with your changes.
You'll find the infos here : https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md#for-pull-request-submitters

bmalynovytch reviewed (changes requested):
You need to add the corresponding documentation, as well as an example.
Good job though 👍

netmonk committed:
[DOC] mysqldb_py

Updating documentation section to explain this new import option.

netmonk committed:
mysqldb_py fixing lint issue

ready_for_review

netmonk commented:
ansibot where are you ?

bmalynovytch commented on lib/ansible/modules/database/mysql/mysql_db.py:
```diff
@@ -117,12 +117,21 @@
+  mysql_db:
+    state: import
+    name: all
+    encoding: "latin1"
```
```suggestion
    encoding: latin1
```

bmalynovytch commented:
@netmonk when you're done with your changes, just type `ready for review` (replace spaces by underscores) in the chat the let the bot know you're done.

netmonk committed:
Update lib/ansible/modules/database/mysql/mysql_db.py

Co-Authored-By: Benjamin MALYNOVYTCH <bmalynovytch@users.noreply.github.com>

netmonk commented:
ready_for_review

bmalynovytch commented on lib/ansible/modules/database/mysql/mysql_db.py:
```diff
@@ -117,12 +117,21 @@
     target: /tmp/dump.sql
 
-- name: Import file.sql similar to mysql -u <username> -p <password> < hostname.sql
+# Basic import of sql script file
```
For conciseness of the examples, maybe should we keep only 1 example with multiple option enabled / shown.
I'd keep only your example

netmonk commented:
ready_for_review

bmalynovytch commented:
@netmonk you marked my review as fixed but it's not and you don't even express your point of view
I won't review further

netmonk commented:
> @netmonk you marked my review as fixed but it's not and you don't even express your point of view
> I won't review further

well, i prefer to have multiple exemples, like there are multiples exemples for others functions of the module. 
Often the Ansible documention is really unclear about usable options when using a module. I prefer to have two exemples with two use-cases specific use case. 
Is it clearer ?

bmalynovytch commented:
The `import` function remains the same.
Specifying the encoding isn't a different function, just an argument.

netmonk commented:
This argument is optionnal, and to be used it needed.

netmonk committed:
fixing documentation

bmalynovytch commented:
LGTM
shipit

netmonk commented:
Do i need to close the pull request now ?

bmalynovytch commented:
No it’ll get closed when merged. Before that other contributors must review and add `shipit` command to confirm they are ok with your code.

bmalynovytch commented:
bot_status

Andersson007 reviewed (commented):
@netmonk , thank you for the PR!
Would be cool to see ci test for this case added to the existing tests here ```test/integration/targets/mysql_db/tasks/```.
Could you please make them.
Then we can merge it without any concerns (IMO ci should be added to all prs except the most primitive and doc related).

netmonk commented:
I will try to do it, need some more time for it. When thinking about it, i just realised that dump is also prone to charset specification. So i can also update the dump function to take `encoding` into consideration. What do you think about ?

Andersson007 commented:
I don’t know :) i can’t think anything because i’m not a hardcore MySQL dba, we use it for a couple of non production services for legacy reasons. So I can look at this only in general. Anyway I believe we need tests for every bugfux/new feature PR not to break anything existent by merging them. Would be cool to be sure that all related options and possible cases that might be affected by the code changes are covered.

netmonk commented:
Im not a dba neither. It's just i just spend the last 6 months advocating against the use of the shell module in Ansible in other dpt of my client company. And then i was asking to automate deployement of a third party software where the mysql scripts are encoded in Latin1 and i had some issue. That is why i decided to look at code and fix it. In order to avoid to use the shell module... :)

Andersson007 commented:
ok, @bmalynovytch approved it, so, if there were ci tests I would approve it too :)

netmonk commented:
will do. gimme time ! :)

mattclay committed:
Remove ansible-test --tox delegation option.

Use the --venv option instead.

This option was only available when running from source to test the ansible/ansible repository.

This will have no effect on testing collections or running from an installed version of Ansible.

Also update docs to reference the --venv option instead of the --tox option.

anusha94 committed:
Resolved merge conflicts

anusha94 committed:
Fixed review comments

Signed-off-by: Anusha Hegde

anusha94 committed:
Merging conflicts and updating PR

Signed-off-by: Anusha Hegde

D3DeFi committed:
re-enable zabbix_host integration tests and add cleanup handlers

atombrella committed:
Replace TestCase.assertEquals with TestCase.assertEqual.

mattclay committed:
Switch from RHEL 8.0 & 8.1b to 8.1 in CI.

sbulage committed:
Fixed few typo's: "Mutiple" to "Multiple" (#64543)

Signed-off-by: Satyajit Bulage <sbulage@redhat.com>

sky-joker committed:
Zabbix: zabbix_template: fix https://github.com/ansible/ansible/issues/64030 (#64032)

add zabbix_template changelog

Andersson007 committed:
postgresql_subscription: improve logic (#64561)

mkrizek committed:
botmeta sanity: check migrated_to on non-existing file (#64551)

modular-magician committed:
Bug fixes for GCP modules (#64486)

modular-magician committed:
Bug fixes for GCP modules (#64485)

modular-magician committed:
Bug fixes for GCP modules (#64484)

n0trax committed:
Replace clear() method for backward compatibility. (#64504)

mattclay committed:
Disable failing aws_acm integration test.

bincyber committed:
Added support for pagination to the github_deploy_key module(#36876)

- Fixes #36745 
- removed json import
- removed requirements from documentation
- refactored the module
- added changelog fragment

Andersson007 committed:
postgresql_publication: fix typo in module.warn (#64582)

* postgresql_publication: fix typo in module.warn

* postgresql_publication: fix typo in module.warn, add changelog

Andersson007 committed:
postgresql_subscription: fix typo in module.warn method name (#64583)

* postgresql_subscription: fix typo in module.warn method name

* postgresql_subscription: fix typo in module.warn method name, add changelog

Akasurde committed:
Correct usage of API call in aws_config_aggregator (#64581)

update_resource and delete_resource takes and requires four argument.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

D3DeFi committed:
zabbix_action shouldn't require args other than name when state=absent

justjais committed:
PR raised to add the feature of adding/removing IP from a Host record (#50975)

* 50238 fix

knagamin committed:
Fixed tiny typo in interpreter_discovery.rst (#64608)

Added missing '/'

jillr committed:
route53_info max_items and next_marker should be a str (#64617)

* route53_info  max_items and next_marker should be a str

Type was changed in #64358, however boto3 takes a str for this option
Add some basic tests for route53_info.
Fixes: #64534

samccann committed:
clarify acceptable file types in collection docs (#64618)

* clarify acceptable file times in collection docs

* implement feedback

Fred-sun committed:
Format azure_rm_keyvault_key_info module's document (#64580)

Fred-sun committed:
Format azure_rm_snapshot document (#64579)

tima committed:
Remove deprecated kubernetes and oc module "indirection" in v2.9 changelog doc (#64572)


+label: docsite_pr

Fred-sun committed:
Format azure_rm_gallery* relate document (#64541)

Fred-sun committed:
Update azure_rm_batchaccount's document (#64490)

Fred-sun committed:
Update azure_rm_azurefirewall relate document (#64488)

jborean93 committed:
Add note about kerberos dependency versions (#64478)

jborean93 committed:
Add warning about winrm on MacoS (#64472)

al-x committed:
minor typos and style changes (#64386)

Glandos committed:
Fix cache example (#64600)

ssbarnea committed:
Documents storage of task files (#62345)

xorel committed:
one_vm: Fix creating VM also for template_id=0 (#63579)

mariolenz committed:
VMware: Implement updates for dvPG in vmware_dvs_portgroup (#62537)

* VMware: Implement updates / changes for dvPG in vmware_dvs_portgroup
* Remove unnecessary else
* Add more integration tests
* Improve create_vlan_list() helper function

kakurpiel committed:
Update aruba modules to handle prompt and syntax changes (#54261)

* Update prompt matching to handle changes introduced in arubaos 8.x

* Update aruba modules to support command changes on arubaos 8.x

felixfontein committed:
docker_login: allow to function with credential helpers (#63158)

* Allow docker_login to function with credential helpers.

* Improve various details and update docs.

felixfontein committed:
docker_swarm_service: Allow source to be omitted for tmpfs mounts (#64637)

* Allow source to be omitted for tmpfs mounts.

* Add changelog.

nkshrishail committed:
nxos_overlay_global: Removing provider and correcting relative references (#64454)

chrisvanheuveln committed:
nxos_igmp_interface: fix regression test failure (#64286)

nkshrishail committed:
nxos_lacp_interfaces: Updating the tests to handle platforms that don't support - lacp mode delay (#64123)

* nxos_lacp_interfaces: Updating the tests to handle platforms that dont support - lacp mode delay

* nxos_lacp_interfaces: Updating search string to include all 9500 series switches

* nxos_lacp_interfaces: Adding a comment about chassis_type

nkshrishail committed:
nxos_lacp: updated tests to handle platforms not supporting lacp system mac command (#64074)

* Updated nxos_lacp tests to handle platforms not supporting lacp system mac command

* nxos_lacp: addressing comments

* nxos_lacp: Updating image tag search to include more tags

kustodian committed:
Remove kustodian from postgres and opennebula (#64647)

I currently don't have time to follow the changes in Postgresql and OpenNebula modules.

Tomorrow9 committed:
VMware: add new module vmware_guest_tools_info (#62860)

Qalthos committed:
Refine default 401 handling (#64619)

sivel committed:
Sort test/sanity/ignore.txt (#64672)

rene1977 committed:
- add nextToken support (#64598)

* - add nextToken support

* - add changelog fragment

* - use boto3 paginator for describe_log_groups

modular-magician committed:
New Module: gcp_serviceusage_service_info (#64589)

modular-magician committed:
New Module: gcp_serviceusage_service (#64588)

modular-magician committed:
Bug fixes for GCP modules (#64587)

felixfontein committed:
Buypass now has a ACME v2 production endpoint. (#64649)

ruzickap committed:
Changed with_items -> loop (#62995)

chris-kiick-sp committed:
archive module documentation is misleading (#62755)

bcoca committed:
really clarify environment (#62531)


Co-Authored-By: John R Barker <john@johnrbarker.com>

sivel committed:
Set alter_sys=True instead of False to address backwards incompat (#64670)

* Set alter_sys=True instead of False to address backwards incompat

* ci_complete

* Add integration test

* ci_complete

* sanity

* ci_complete

* Changelog fragment

* Update import test and validate-modules to match

coryasilva committed:
Add Local Security Policy update example (#60897)

* Add Local Security Policy update example

##### SUMMARY
This real world example will be useful, but perhaps there is a better section?


##### ISSUE TYPE
- Docs Pull Request

+label: docsite_pr

* Move new example and add not to username

jborean93 committed:
win_package - Set raw for argument type (#64690)

kvprasoon committed:
change $partition variable name to $ansible_partition (#63968)

* change $partition variable name to $ansbile_partition

* Added changelog fragment

philseeley committed:
Make the DNS service optional when installing Active Directory (#63440)

* Added install_dns option to the win_domain module.

* Added install_dns option to the win_domain_controller module.

* Updated version_added to devel branch version 2.10.

* Updates for ansible-test errors.

* Minor nits to docs

haiyuazhang committed:
Fixes azure_rm_snapshot (#64542) (#64547)

* feature enhancement: make a snapshot from an existing managed disk.

* remove default creation option in argu_spec of azure_rm_snapshot

* fix pep8 related issue

* fixing broken integration test

h-hirokawa committed:
Add encoding and codepage params to win_command/win_shell (#54896) (#54966)

* Add output_encoding_override params to win_command/win_shell (#54896)

This enhancement enables Ansible to parse the output of
localized commands that ignore the prompt code page.

* Added changelog and minor nits

kvprasoon committed:
Add output options for win_pester (#63583)

* add output options for win_pester

* fix typo

* correct variable name

Akasurde committed:
doc: Handle exception in parsing parameter description (#60933)

Fixes: #60587

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
VMware: Handle slash in network name in vmware_guest module (#64494)

Encode slash in network name to work with vSphere API.

Fixes: #64399

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

samccann committed:
remove toshio as release manager (boohoo) (#64685)

D3DeFi committed:
added missing atexit logout for zabbix_host module (#63785)

Qalthos committed:
Fix eos_facts over local eapi (#64570)

* Fix eos_facts over local eapi

* Much better feedback when querying resources over local

* No need for that anymore

* Update comment

* Mkae sure facts are tested on local

resource modules might break this otherwise

* This test was not checking anything

* Pass over nxos_facts as well

Akasurde committed:
Remove temp directory created by wait_for_connection (#64592)

wait_for_connection creates AnsiballZ_ping.py in temp directory,
which remains on remote machine even after playbook run.

Fixes: #62407

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

pilou- committed:
command, script & shell: mention 'cmd' parameter (#62778)

'cmd' parameter is supported since 2b3c5aa41ed135d4ba0f125670ca4bd048149679 (v2.0.0-1)

jiuka committed:
win_domain_user: Make Identification of the user to work with more robust (#61594)

* Ensure we work on only one user.

After the initial get/create use the GUID of the found/created user to
ensure we will not start to work with  a different user. If we create a
user or modify it's attributes an he is not identified with the name
parameter afterwards this module fails in rather unpredictable ways.

This addressed #45298

* Use splatting create_args for creating user.

This prepars this for adding more optional create arguments without 
nesting of condictions.

* Set the UserPrincipalName and SamAccountName on create.

Set the UserPrincipalName and SamAccountName on the create operation if 
upn is given to ensure the User is created with a contollable 
SamAccountName.

* Rename $username to $name.

$username is missleading as its not the SamAccountName.

* Add a identity parameter to win_domain_user

This gives the user full controll over how the user is identified in the 
AD.

* Add version_added information for new parameter and fix yaml syntax.

* Added changelog fragment

ShachafGoldstein committed:
setup.ps1 - parity with linux regarding missing local facts path (#58887)

* setup.ps1 - parity with linux regarding missing local facts path

* add warning

* Update setup.ps1-parity-with-linux-regarding-missing-local-facts-path.yml

sergioirium committed:
Corrected Get-adcomputer (#63093)

* Corrected Get-adcomputer

Corrected Get-adcomputer on "Remove-ConstructedState" and "Set-ConstructedState" functions.
resolved error: Unable to contact the server. This may be because this server does not exist, it is currently down, or it does not have the Active Directory Web Services running.

* Update lib/ansible/modules/windows/win_domain_computer.ps1

Co-Authored-By: Daniel-Sanchez-Fabregas <33929811+Daniel-Sanchez-Fabregas@users.noreply.github.com>

* Update win_domain_computer.ps1

changed
"-credential $credential" to "@extra_args" (Line 115 and 150)
corrected exception message (Line 122)

* Added changelog fragment, minor code tweak

P-Hessler committed:
win_acl no longer needs SeSecurityPrivilege (#57804)

* win_acl no longer needs SeSecurityPrivilege
Set-ACL raises missing SeSecurityPrivilege error when the inheritance
from the parent directory is disabled.

* fixes test sanity

* registry rights can only be modified with Set-ACL

* add changelog

artazar committed:
Improve handling of "restarted" state for win_iis_website module (#63829)

* Improve handling of "restarted" state for win_iis_website module

* Added changelog fragment

Commit:
vmware_tools handle connection issues (#57661)

* Catch vmodl.fault.SystemError exception.

* Add exceptions for connection issues.

* Fix self._si not defined exception (L353), when trying to close a non existing connection.

* Fix vmware_tools connection reset

* Also catch InvalidGuestLogin exception, as it get's also thrown when trying to logging in too early on startup...

chillancezen committed:
Fortinet's new module for fortios_vpn_ipsec_phase2_interface (#64767)

richlv committed:
Update openssl_certificate.py (#64772)

Changed 'generate' to 'generating'

GomathiselviS committed:
eos_lacp_interfaces : Fix to - lacp port-priority not set when… (#64530)

* Fix bug 64453

goneri committed:
vmware/test: use better name for datastores

In the VMware tests, we call the datastores `ds1` and `ds2`. The first
one is read-only, the second is read-write and can be used to deploy
VMs. The naming convention was not clear enough and source of a lot
confusion and mistake.

We now have two better names:

- ro_datastore, which is ... read-only
- rw_datastore, the one that we can use to deploy new VM.

goneri committed:
vmware_guest: fix the name of the test VMs

The test VM names must be deterministic, this way, the teardown playbook
can clean up the resources after the test execution.

See: https://docs.ansible.com/ansible/latest/dev_guide/platforms/vmware_guidelines.html#vm-names-should-be-predictable

mjlshen committed:
[FIX] graceful is not an immutable attribute (#64397)

jamescassell committed:
Document match/search tests (#64775)

* clarify `search` vs `match` tests

* string test docs: fix rst formatting

verasativa committed:
docs: fix gcs example to work with ansible 2+ (#64778)

switched directive sudo to become

jamesv1994 committed:
update documentation example (#64786)

missing k in workflow in an example for the tower_workflow_template module

jpmahowald committed:
Syntax fix guide_gce.rst (#64779)

Fix syntax for create a GCP instance example playbook. Fix task indentation.

vincentbernat committed:
ipaddr: add an option to return peer of a point-to-point link (#59144)

* ipaddr: add an option to return peer of a point-to-point link

Signed-off-by: Vincent Bernat <vincent@bernat.ch>

* ipaddr: extend "peer" to also work with /30

jborean93 committed:
Add the ability to ignore files and collection build (#64688)

goneri committed:
vmware_guest_disk/test: use less disk space

Reduce the amount of space that is used on the datastore during the
test. This allow use to use lighter storage environment during the test.

tremble committed:
ec2_eip Don't throw an exception when re-releasing an EIP (idempotency) (#62332)

* ec2_eip: (integration tests) move to using module_defaults

* ec2_eip: (integration tests) expand integration tests

Also clean up a little
- Delete EIPs when we finish testing them (reduce the chance of hitting limits)
- Rejig deletion so that it works when runs fail
- Add tests for ec2_eip_info

* ec2_eip: Minor doc tweaks

* ec2_eip: Don't throw an exception when we try to disassociate an already disassociated EIP

* ec2_eip: Add missing IAM policy (manage IGWs)

* ec2_eip: (integration tests) Use the VPC as a crude lock to avoid running parallel tests

We test that untagged EIPs come and go as we expect, if multiple tests are
running in parallel this confuses things

* Fix ec2_eip association

tchia04 committed:
allow using --check on win_iis_webapppool module (#50528)

* allow using --check on win_iis_webapppool module

* Added changelog and slight logic tweak

* Fix typo in changelog fragment

dlitster committed:
Update win_firewall_rule.py to include ICMP echo (ping)  example (#55013)

* Update win_firewall_rule.py

<!--- Your description here -->
Added example of enabling ICMP protocol, as ping is commonly used for troubleshooting in automation scenarios.  Equivalent netsh command is: 
netsh advfirewall firewall add rule name='ICMP Allow incoming V4 echo request' protocol=icmpv4:8,any dir=in action=allow

+label: docsite_pr

* Remove yaml-breaking space

Removed extraneous space that caused validation to fail.

* Remove more extraneous whitespace

ShachafGoldstein committed:
win_firewall_rule - support ICMP types and codes (#60361)

* win_firewall_rule - support ICMP types and codes

* Update main.yml

* Add more examples and change type to list

Daniel-Sanchez-Fabregas committed:
win_domain_computer module:  Fix idempotence when name != sam_account_name  (#56967)

* win_domain_computer module: Minor documentation error

* Fix idempotence when name != samaccountname

* Added changelog

* Added unsupported alias for CI check

YouhuaLi committed:
use ~/.one/one_auth as default of ONE_AUTH file location (#64535)

* better excpetion handling when one_auth file is missing
* use Exception instead of BaseException

jtyr committed:
VMware: Simplifying get_vm() in module_utils.vmware (#62217)

aisbergg committed:
hostname - Add Manjaro Linux distribution (#64810)

* Add Manjaro Linux distribution

Cohedrin committed:
Fix postgres_user not commiting changes when groups is set (#64807)

resolves #64806

Akasurde committed:
gitlab_user: Remove optional parameters while state is absent (#64832)

While deleting gitlab user, several parameters such as email, name,
password are not required. This fix removes this requirement from the
module.

Fixes: #61921

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

robertdebock committed:
Add advice how to prevent jinja2 warning (#60594)

* Add advice how to prevent jinja2 warning

##### SUMMARY
This addition helps users to demonstrate how to prevent `[WARNING]: when statements should not include jinja2 templating delimiters`.


##### ISSUE TYPE
- Docs Pull Request

+label: docsite_pr

* Update playbooks_tests.rst

* Update playbooks_tests.rst

Refer to the advised FAQ about when to use `{{ }}`.

mkrizek committed:
Fix ansible_failed_{task,result} undefined in rescue (#64831)

This is a fix for a regression introduced by Perfy. Since then we mainly
operate on host.name instead of the Host object. In a call to
set_nonpersistent_facts where we set ansible_failed_task and
ansible_failed_result variables we were still passing the object which
led to those vars being undefined.

Fixes #64789

t-reppert committed:
Adjust conditional for disabling alert email (#63951)

* Adjust conditional for disabling alert email

Conditional was improperly disabling existing alert email when it was found enabled.  The expectation is that it should only disable if module.param['enabled'] was set to false.

* Adjusted disable logic

Added back in the logic of only disabling the alert email if it is module.params['enabled'] is false AND current value of enabled is true on the array

jasonpepper committed:
Fix kinesis_stream wait loop pause. (#64283)

Commit:
Update RDS snapshot info to use DBClusterIdentifier (#64253)

JulienPalard committed:
Close subprocess stdout and stderr in the ssh connection plugin

Eliminate `ResourceWarning: unclosed file` by closing all
file handles of each subprocess used by the SSH connection
plugin to execute commands.

This change prevents Ansible from accumulating "forgotten"
open file handles.

PR #64785 by Julien Palard
Fixes #64768

prasadkatti committed:
Add a random component to state machine name in aws_step_functions_st… (#64476)

* Add a random component to state machine name in aws_step_functions_state_machine testsuite

* Pick a random number from a bigger set to avoid conflicts between parallel runs

Co-Authored-By: Matt Clay <matt@mystile.com>

anusha94 committed:
VMware: new module vmware_guest_serial_port (#63126)

The ask was to add an option in the vmware_guest itself. Upon inspection, a serial port can
be created in multiple ways and so can be a module in itself. Therefore, created a new module
called vmware_serial_port.

Fixes: #54956

johnbui423 committed:
Default to python3 on Debian 10 (#63097)

tremble committed:
ec2_vpc_net_info: integration tests (#62649)

* ec2_vpc_net: (integration tests) migrate to using module_defaults

* ec2_vpc_net: (integration tests) use a private subnet for the tests

* ec2_vpc_net_info: Add integration tests

* ec2_vpc_net_info: add cidr_block_association_set to documentation

* Update AWS hacking test policy to allow VPC CIDR disassociation

* Update test/integration/targets/ec2_vpc_net/tasks/main.yml

Co-Authored-By: Jill R <4121322+jillr@users.noreply.github.com>

* Store vpc2 ID to make it clearer which VPC we're changing

* Be more consistent with our quoting

* Explicitly test that the VPC IDs haven't changed

faridda committed:
Add default value to 0 for disk in nova_flavor module (#64749)

* Add default value to 0 for disk in nova_flavor module

* Apply suggestions from code review

Add type int for disk size

Co-Authored-By: John R Barker <john@johnrbarker.com>

srivignessh committed:
Fix access variable name programmatically faq docs (#64041)

* Add an example of replacing dashes with underscores in interface names for IPv4

Gianlu committed:
Modified setup.ps1 in order to show Windows Installation type (#41002)

* Modified setup.ps1 in order to show Windows Installation type

* Fix after pull request comment

* Added changelog fragment

Daniel-Sanchez-Fabregas committed:
win_domain_user: add retry logic for null user principal group (#54334)

* win_domain_user: add retry logic for null user principal group

* win_domain_user.ps1: Fix "user without group" case use

* Added changelog fragment

* Fix up missing dollar sign

dagwieers committed:
win_chocolatey: Fix error when choco.exe not found (#53115)

* win_chocolatey: Fix error when choco.exe not found

* Slight tweak to check and added changelog fragment

* Removed ignore rule that's no longer needed

jborean93 committed:
win_pester - Align dir scanning with Pester's defaults (#64880)

* win_pester - Align dir scanning with Pester's defaults

Co-authored-by: Coby Caldwell <cobycaldwell@gmail.com>

* Fix sanity issues

wilmardo committed:
win_disk_facts: Adds Win32_DiskDrive object to facts (#51595)

* Adds win32_disk_drive object to win_disk_facts

* Names class parameter for Get-CimInstance as requested in the devdocs

* Maps whole class and adds docs

* Improve matching of disks when UniqueID is different format

* Improve logic for PNPDeviceID mapping

* Adds test for win32_disk_drive

mhunsber committed:
win_iis_webapplication: add authentication parameters (#56033)

* add connect_as, username, password parameters
add tests

* fixed reference to undefined variable.
added version added to new options.

* add changelog fragment

* fix line endings

* use ansible facts to determine os version
remove unused iis version check
test checksum of iis configuration after backup

* correct assertion

* added more cleanup tasks.

* version added is now 2.10

* skip server 2008 r2 for now

* run tests on server 2012 and higher

Andersson007 committed:
postgres.py: add query_params (#64661)

* postgres.py: add query_params

* postgres.py: add query_params, add changelog

sbulage committed:
Improved message for unarchive. (#64594)

Signed-off-by: Satyajit Bulage <sbulage@redhat.com>

demizer committed:
Add timeout to jenkins_script csrf requests (#62850)

anupamaloke committed:
Add support for multiple manager attributes configuration (#60201)

* Add support for multiple manager attributes configuration

* Updated version_added to 2.10

* Added support for modifying LC and system attributes

* Added support for LC and system attributes

* Pylint fixes

* Add support for multiple manager attributes

* Add support for multiple manager attributes

* Add support for multiple manager attributes

Andersson007 committed:
postgresql_subscription: fix the module can return incorect changed value when owner is passed (#64885)

goneri committed:
module_defaults: add tree missing vmware modules (#64877)

The following modules a new, and were not yet in `module_defaults.yml`:

- vmware_guest_register_operation
- vmware_guest_serial_port
- vmware_guest_tools_info

samccann committed:
update configuring ansible-galaxy client (#64796)

* update configuring ansible-galaxy client, deduplicate, clarify tokens, AH procedures, etc.

samdoran committed:
Fix firewalld integration tests for CentOS 8 (#64873)

samdoran committed:
Fix setup_postgresql integration test for CestOS 8 (#64871)

samdoran committed:
Fix mysql_replication integration test for CentOS 8 (#64870)

samdoran committed:
Fix setup_mysql8 tests for CentOS 8 (#64868)

# Conflicts:
#	test/integration/targets/firewalld/tasks/main.yml

samdoran committed:
Fix filesystem integration test for CentOS 8 (#64866)

samdoran committed:
Fix mariadb_replication tests for CentOS 8 (#64865)

samdoran committed:
Fix setup_mysql_db for CentOS 8 (#64864)

samdoran committed:
Fix yum_repository tests for CentOS 8 (#64863)

Refactor tests to run the same tasks with CentOS and Fedora using different variables.

samdoran committed:
Fix MariaDB integration tests for CentOS 8 (#64859)

* Fix setup_mariadb for CentOS 8
Switch to using systemd for multiple instances on CentOS 7 and 8

samdoran committed:
Fix group integration test for CentOS 8 (#64858)

samdoran committed:
Fix setup_rpm integration tests for CentOS 8 (#64862)

samdoran committed:
Fix paramiko_ssh test (#64857)

samdoran committed:
Fix iso_extract tests for CentOS 8 (#64861)

- use setup_epel role rather than task which contains correct URL now since it has changed for EL8
- add conditional to setup_epel to prevent attempted installation on non-RHEL distros

samdoran committed:
Fix lookup_passwordstore integration test for CentOS 8 (#64869)

pass is no longer packaged in EPEL
https://lists.zx2c4.com/pipermail/password-store/2019-July/003689.html

yuandongx committed:
Revert "Doc notes 4 (#64349)" (#64922)

This reverts commit 703f7a149ac882712180f4c78683a11ab54c635f.

yuandongx committed:
Revert "Doc notes 5 (#64350)" (#64921)

This reverts commit 1d40d429a0397e2a3799932298a11c27598fc588.

sky-joker committed:
VMware: new module vmware_host_auto_start (#64605)

Fixes: #63905

* Add return parameters to return after module execution https://github.com/ansible/ansible/pull/64605#discussion_r344430717
* Add moid and use_instance_uuid parameters https://github.com/ansible/ansible/pull/64605#discussion_r344431003
* Add reset playbook for test case
* Delete HAS_PYVMOMI variable https://github.com/ansible/ansible/pull/64605#discussion_r344430877
* Fix because the default value of stop_action was wrong
* Fix multiple sanity test error for document related
* Fix remove unnecessary line break

mossroy committed:
Fix argument name in example of documentation (#64891)

##### SUMMARY
The second example of the documentation uses an argument "hostname" that does not exist in this module.
It should be replaced by argument "name"


##### ISSUE TYPE
- Docs Pull Request

+label: docsite_pr

yuandongx committed:
Doc notes 4 (#64927)

* addd

* all  update

* add/update notes.

* revert and exclubde new in 2.10

yuandongx committed:
Doc notes 5 (#64928)

* addd

* all  update

* add/update notes.

* add/updates notes.

* new version 2.10 can not be backported

yuandongx committed:
add notes for 4 ce moudules docs. (#64929)

* add notes.

* duplication of key "notes" in mapping

mariolenz committed:
VMware: New module vmware_host_dns (#64458)

* Restore module vmware_dns_config
* Remove domainname and change_hostname_to
* Changed version_added from 2.10 to '2.10'
* Add setup_attach_host: true to test case
* Add 'vcsim is not defined' block to integration tests
* Change 'result' to 'dns_config_result'
* Bugfix: Changing some static configurations while keeping others can crash the module
* Implement changing DNS config from DHCP to static on a cluster
* Update documentation for vmware_host_dns
* vmware_host_dns integration tests: Always revert to original DNS configuration, even if a test fails
* Deprecate vmware_dns_config

drybjed committed:
Add module ldap_attrs; deprecate ldap_attr (#62701)

* Add ldap_attrs module

* Fix codesyle error

* Apply deprecation rules

* Add support for X-ORDERED extension in ldap_attrs

The 'X-ORDERED' LDAP extension allows definition of ordered LDAP object
attributes. This extension is used in OpenLDAP "cn=config" database to
support ordered configuration options.

Specification: https://tools.ietf.org/html/draft-chu-ldap-xordered-00

* Update ldap_attr deprecation notice

* Documentation improvements based on suggestions

* Remove redundant dots from documentation

* Correct 'insertations' to 'insertions'

* Remove insecure 'params' option

* Fix sanity ignore errors

* Improve module documentation

* Change example value from string to list

* Fix support for "" values

Akasurde committed:
VMware: Re-introduce backing_disk_mode in vmware_guest_disk_in… (#63244)

This info about backing_disk_mode was renamed to backing_diskmode,
to keep backward compatability re-introducing backing_disk_mode along
with backing_diskmode info.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

samdoran committed:
Use newer version of podman on RHEL (#64934)

- Use same podman version for RHEL 7 and 8
- Pin to version 1.4

jborean93 committed:
become - stop using play context in more places (#62373)

* become - stop using play context in more places - ci_complete

* Fix up review points

agowa338 committed:
Add full IPv6 support to win_dns_client - Fixes #55962 (#57577)

* Add full IPv6 support to win_dns_client - Fixes #55962

* Fix missing cast

* Add type to win_dns_client.py

* Remove version_added again, to hopefully make ansibot happy. Even though it was added as a response to the bot...

* Fix $params undefined error, that was introduced by fixing the "global variable" linting issue

* Fix casting error

* Fix inverted logic

* Fix rebase error

* Fix assignment to readonly variable

* Fix "reset IPv4 DNS back to DHCP adapter_name"

* Fix legacy windows server support (2008/2008R2)

* Fix 2k8

* Remove unecessary pslint ignore

* Added IPv6 tests, changelog fragment and further docs

jborean93 committed:
Correct win_dns_client test option name (#64967)

rrey committed:
Move grafana modules to own subdir (#64657)

markchalloner committed:
Fix Grafana lookup auth header for py2/3 compatibility (#50816)

mwl committed:
hcloud: inventory: missing compose variables (#64559)

tremble committed:
Update Route53 IAM policy so the Route53 tests run (#64886)

acozine committed:
Collections versions (#64912)

samdoran committed:
Fix postgresql_lang integration test for CentOS 8 (#64872)

* Only search inside the role for files to include
When a role is called as a dependency, both roles are searched. This can cause the incorrect file to be included if it is picked up from the parent role.

rzerda committed:
Fix typo in grafana_datasource docs (#65000)

bmillemathias-1a committed:
Remove duplicated and broken link to Roles (#64991)

bmillemathias-1a committed:
Fix link to Roles (#64992)

Commit:
renamed module tls client auth params to avoid overlaping with fetch_url (#59522)

* renamed module tls client auth parameters to avoid overlaping with ansible fetch_url

* added version_added info for params

* Updated version_added

Updated version_added info from 2.9 to 2.10

* Update pulp_repo.py

removed version_added for renamed params

* Apply suggestions from code review

added ca_cert alias and 'version_added'

Co-Authored-By: Mark Chappell <mchappel@redhat.com>

* Apply suggestions from code review

added old behavior for client_cert and client_key which will deprecate in 2.14

Co-Authored-By: Mark Chappell <mchappel@redhat.com>

* Update pulp_repo.py

fix for linting error ansibot is complaining

* added changelog fragment for 59522

* Apply suggestions from code review

more informative depreciation warning and changelog fragment

Co-Authored-By: Felix Fontein <felix@fontein.de>

* added mention for changes in client_key and client_cert behavior

* fixed too long line (linting)

* deprecated ca_cert alias to have consistent module params in Ansible 2.14

* fixed indentation for deprecation warning

* changed deprecated alias handling to argument_spec

* moved deprecated_aliases insied argument dict, thanks tremble

* suggestions from felixfontein

Move doc info about client_cert and client_key into its own paragraph

Co-Authored-By: Felix Fontein <felix@fontein.de>

samccann committed:
fix collection broken links (#65010)

Akasurde committed:
distribution: ignore plesk-release file while parsing (#64665)

/etc/plesk-release file is now ignored while parsing distribution
release.

Fixes: #64101

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

samdoran committed:
Fix timezone integration test for Fedora 31 (#65013)

samdoran committed:
Make Docker and Podman play nice (#65009)

podman 1.4 now installs docker man pages. Add handler to remove docker packages to avoid test failure due to this conflict.

Adjust inventory_docker_swarm integration test
Add conditional to cleanup handlers to allow use of the setup_docker role without removing packages at the end of the play. The inventory_docker_swarm integration tests does its own cleanup.

tremble committed:
Feature: iam_role tags (#63944)

* iam_role tags support

* Make sure we don't Camel -> Snake tags in our return values

* Minor documentation tweaks

* Add tagging tests

* Make sure we return the state of tags once we updated them

* Update lib/ansible/modules/cloud/amazon/iam_role.py

Co-Authored-By: Felix Fontein <felix@fontein.de>

* Move boto3/botocore before we start making changes

s-hertel committed:
Update cache plugin documentation (#63635)

* Update cache plugin documentation since all cache plugins shipped with Ansible can be used for caching inventory since 2.8

* Update docs/docsite/rst/plugins/cache.rst

Co-Authored-By: Abhijeet Kasurde <akasurde@redhat.com>

Co-Authored-By: John R Barker <john@johnrbarker.com>

toadjaune committed:
Improve examples on parameters use with argv (#60781)

* Add example showing that when using the `argv` syntax for command module instead of the string one, we can directly provide other parameters in the "command" block, rather than using an "args" block.

prasadkatti committed:
cloudformation_info Move sample outputs from EXAMPLES to RETURN section (#64298)

Tomorrow9 committed:
VMware: Fix wait for customization timeout issue (#64493)

chrisvanheuveln committed:
nxos_vlans: fix rmb behaviors and tests (#63650)

* nxos_vlans: fix rmb behaviors and tests

* whitespace fixes

* whitespace fixes 2

* review comments addressed

* TBD for legacy support

* additional tests to hit code coverage misses

* whitespace

chillancezen committed:
Fix py2/py3 compability issue for httpapi plugin fortios (#64982)

Lunik committed:
gitlab modules: unify authentication method (#64989)

* gitlab modules: unify authentication method
* gitlab module utils: remove useless import
* gitlab module utils: add documentation
* gitlab module utils: use distutil to compare version
* gitlab modules: sanity
* gitlab modules: remove useless import
* gitlab module utils: add missing import
* add changelog

Akasurde committed:
Add missing version_added in checkpoint documentation (#65048)

Domain is newly added in 2.10 release, specify this in the
documentation section in Checkpoint httpapi connection plugin.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
vmware_guest: Revert custom value customization in guest (#65031)

Due to confusion between 'Custom Attributes' and 'Advanced options',
this change got in devel. Revert to original behavior i.e. customizing
custom values using vmware_guest is done in this PR.

Fixes: #64291

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

yuandongx committed:
Add doc to ce for user guide.  (#63982)

felixfontein committed:
Document diff. (#64888)

samdoran committed:
Update ansible.cfg (#64855)

- clean out old options that are now deprecated
- clean up formatting of comments a bit
- add become plugin examples

Bjoern2 committed:
Fixes error in regex example (#65047)

tremble committed:
Amazon modules - Yet more minor documentation cleanup (#64930)

urrameu committed:
Fix text in import_role module description (#65066)

Docs: Remove unnecessary "it" in description.

mattclay committed:
Fix ansible-test handling of git submodules. (#65027)

* Revert most of PR #61605 commit e218c9814c3cc8d50a8cab23c2bb69061b3b9be9

This removes the git error handling that converted all git errors into warnings.

* Fix ansible-test handling of git submodules.

bmillemathias-1a committed:
Put all files and Path filters under a specific section (#64913)

Many of the filters in the "other useful filters" list manipulate filenames and paths; add a heading and put them together.

samccann committed:
fix net_system domain_search examples (#65080)

jborean93 committed:
Fix up broken Windows test when run with coverage (#65085)

ShachafGoldstein committed:
win_firewall- Change req check from wmf version to cmdlets presence (#64998)

jborean93 committed:
Windows coverage - output temp files with UTF-8 BOM (#65086)

* Windows coverage - output temp files with UTF-8 BOM
ci_complete ci_coverage

* Remove testing changes to get ready for merge

jborean93 committed:
config manager str type vault source (#65023)

* config manager str type vault source

* Convert vault text to_text earlier and add tests

FreedomBen committed:
Fix typo in docs for lvol module(#65092)

Changed activate to active

tcraxs committed:
fix module that it reports the correct status of a change (#65061)

chkp-orso committed:
fix all modules to be able to get parameter with value false (very important fix) (#65040)

* add import of future and __metaclass__

* fix all modules to be able to get parameter with value false (very importent fix)

* Update ignore.txt

Akasurde committed:
VMware: Find unique tag using category id (#61082)

If two tags with same name and different category exists, vmware_tag_manager
used to take first found tag.

This commit use combination of tag and category to identify the category.

Fixes: #59379

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

rchikatw committed:
Handle gfid tag when producing ansible facts in gluster_heal_info (#65041)

we ignore lines beginning with / or < which will produce the requires json format.

Fixes: #65095

ganeshrn committed:
Add support for jumphost setting in junos_scp and junos_package (#64086)

* Add support for jumphost setting in junos_scp and junos_package

* Since junos_scp and junos_package module uses junos-eznc library
  to transfer file and load package respectively to a remote device
  it does not read the jumphost related configuration in netconf
  connection plugin unlike other junos modules which uses the
  Ansible persistent.
* Add `ssh_config` and `ssh_private_key_file` to add support to
  read custom ssh config file and mention ssh private key file
  for junos_scp and junos_package module

* Fix CI issue and update note section

kbreit committed:
Meraki - Remove integration tests which work with HTTP respons… (#64975)

- These are modular level and I'm not catching these anyways
- More appropriate for unit tests, at least for HTTPS

yuandongx committed:
add a new module to configure isis on isis-view. (#63720)

* add a new module.

* update docs.

* update docs.

* update again.

* add __future__

* update docs, some args type

* update docs, some args type

* add integration test.

* add unittest.

* update for shippabel.

* update for syntax error.

* Update setup.yaml

* Update setup.yaml

* Update cleanup.yaml

* update for review.

* add notes.

Andersson007 committed:
postgresql_lang: use query parameters with cursor.execute() (#65093)

* postgresql_lang: use query parameters with cursor.execute()

* add changelog fragment

felixfontein committed:
openssh_keypair: fix idempotence issue (#65017)

* Fix idempotence issue.

* Add changelog.

billdodd committed:
DeleteUser: do not disable user if already disabled (#64797)

* DeleteUser: do not disable user if already disabled

* add changelog fragment

panyy3 committed:
add new command SetManagerNic in redfish_config module (#64477)

* add new command SetManagerNic in redfish_config module

* use a more explicit/rigorous way to select the EthernetInterface, split port for default nic_addr if root_uri has port, update variable name to lower_case_with_underscores instead of CamelCase

* add missing whitespace around arithmetic operator, fix inline comment should start with '# '

sky-joker committed:
Zabbix: new module zabbix_template_info (#64027)

Remove unnecessary module import(LooseVersion)

Im0 committed:
rabbitmq_publish SSL certificate implementation (#62036)

* Implementing ability to specify certificates.

* Changelog fragment for rabbitmq_publish certificate checking

* Fixing version_added

* Reducing line size.

* Minor documentation updates.

* Update to add missing space.

Co-Authored-By: Felix Fontein <felix@fontein.de>

nrwahl2 committed:
Fixes #28325: Add ansible_ssh_host_key_<algorithm>_public_subtype fact for each public host key algorithm (#28449)

yuandongx committed:
Update plugins/netconf/ce.py to fix a bug(failed to get veriosn information via neocnf). (#65065)

* fix version fail.

* add a changelog fragment.

* Update ce.py

* Rename plugins-netconf-ce-fix.yaml to 65065-plugins-netconf-ce-fix.yaml

* Update ce.py

atombrella committed:
Remove uneeded Python version compatibility checks. (#64076)

samdoran committed:
postgresql_privs - fix sort comparison (#65125)

The result of .sort() is None, not the sorted object. The comparison was comparing the result of the .sort() method and not the sorted values.

tremble committed:
Rewrite iam_policy using boto3 (#63924)

* reworked iam_policy

* Deprecate policy_document option

* deprecate defaulting skip_duplicates to true

* No longer explicitly catch ParamValidationError.

ParamValidationErrror is already caught by ClientError

* Work with complex policy objects rather than json documents

comparisons can better cope with the special cases (eg True vs "True" )

* Enable check_mode tests and fix related 'changed' bug

* changelog

* doc cleanup based on review

nillvitor committed:
Update influxdb_write.py (#65117)

Remove duplicated line

D3DeFi committed:
Added missing logout to zabbix_user_info module (#65146)

gvsheva committed:
fix utils.color.stringc: enclosure non printable sequences in SOH,STX (#64751)

* ansible.utils.color.stringc: add wrap_nonvisible_chars flag in stringc
* add exaplanation for `wrap_nonvisible_chars` case in utils.stringc
* add changelog entry

Miouge1 committed:
Add support for all_projects in OpenStack inventory module (#55965)

felixfontein committed:
#59522: adjust messages (#65129)

* Adjust messages.

* Fix one more.

mjmayer committed:
Add aws elb multiple host header support (#65021)

* Add support for aws elbv2 multiple host header actions

* Add example of multiple host header

* Add test for multiple host headers

* Add missing quote

* Update test/integration/targets/elb_application_lb/tasks/test_modifying_alb_listeners.yml

Co-Authored-By: Jill R <4121322+jillr@users.noreply.github.com>

* Update test/integration/targets/elb_application_lb/tasks/test_modifying_alb_listeners.yml

Co-Authored-By: Jill R <4121322+jillr@users.noreply.github.com>

* Remove debug statements

acozine committed:
link from quickstart page to Network Getting Started (#65150)

heuels committed:
docs: add solution to timeout errors to docs (#65140)

felixfontein committed:
module sanity checks: improve alias handling (#59060)

* add_file_common_args is only of interest on top-level.
* Handle undocumented arguments in one place.
* Update ignore.txt
* Add changelog

tremble committed:
Improve consistency of IAM_{role,user,group} behaviour (#64258)

* iam_user: use AnsibleAWSModule.client to fetch connection rather than C&P code

* iam_role: Add deprecation warning so we can switch purge_polices default behaviour from true to false

* iam_user/role/group: Rename 'managed_policy' and 'purge_policy'

Rename from singluar to plural (we accept a *list* of policies), and add aliases for the old values.

* Cleanup documentation

* Changelog

prasadkatti committed:
Add integration tests for aws lightsail (#63770)

* Add integration tests for aws lightsail

* lightsail - use module_defaults instead of aws_connection_info

* lightsail tests - assert instance state on create

* Fix yaml syntax error

Co-Authored-By: Jill R <4121322+jillr@users.noreply.github.com>

* [lightsail] create keypair as part of the testsuite

* Fix lightsail actions in compute-policy

* Add ability to delete keypair in lightsail_keypair

chillancezen committed:
Fortinet's new module for fortios_system_global (#64817)

davedow15 committed:
vmware_deploy_ovf: Use correct datastore in multi-datacenter environment

Fixes: #63920 

Need to pass the datacenter to assure we get the correct datastore managed object Id.

Akasurde committed:
VMware: Fix typo in fail_json in vmware_guest_powerstate (#65162)

Fixed typo from "chanaged" to "changed"

Fixes: #65161

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

nmandy-git committed:
Update notes in vmware_guest about SysPrep Windows Guest (#65055)

NilashishC committed:
Fix legacy facts keys having redundant underscore (#65167)

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

Tomorrow9 committed:
VMware: add new module vmware_guest_controller (#59615)

Add the `vmware_guest_controller` module.

meerkampdvv committed:
VMware: vcenter_folder checks nested parent folder path (#62741)

added possibility to create Subfolder by defining full path below datacenter

Closes: #64442

samdoran committed:
Remove postgresql_shared integration test (#65133)

Move test tasks into appropriate integration test

markuman committed:
elbv2 - print() statement results an task failure, even if the task itself was successfull (#65183)

cbudz committed:
add note re required forward slash on url (#65203)

pgbidkar committed:
Fix issue 64503. Added pool to relocate spec (#64544)

VM relocate to destination host works without pool param when hosts are part of same cluster. but spec.pool is required when hosts are part different clusters.

Closes: #64503

kryptonite303 committed:
user - compare macOS user properties using same type (#62973)

self._get_user_property returns a string, so when doing a comparison
using this value, cast the second variable to a string so that the
comparison behaves correctly
* Add changelog
* Add to_text import
* Add integration test

goneri committed:
vmware_guest: do not use autoselect_datastore in test (#65191)

Disable the use of `autoselect_datastore` until https://github.com/ansible/ansible/issues/58541
is fixed.

goneri committed:
vmware_rest_client: pass object to get_tags_for_dynamic_obj (#65156)

This patch fix the `vmware_cluster_info` functional-tests.
`get_tags_for_dynamic_obj()` expects an object, not the object ID in a
string.

Without this patch, the `vmware_cluster_info` module fails.
`get_tags_for_object()` raises the following exception:

```
vmware.vapi.exception.CoreException: Expected VapiStruct instance or python dictionary, but received str
```

The patch partially reverts changes introduce in:
https://github.com/ansible/ansible/commit/35cc26f8c0447ab1ad4427eafcc7283c4356370d

See: http://vmware.github.io/vsphere-automation-sdk-rest/6.5/operations/com/vmware/cis/tagging/tag_association.list_attached_tags-operation.html

rohit-gohri committed:
[yarn] Set name_version when version is not specified (#62348)

* Set name_version when version is not specified, fix #55097

This will default to installing the latest version available

* Add changelog fragment

* update changelog

mcassaniti committed:
docker_swarm_service: Sort lists when checking for changes (#63887)

* docker_swarm_service: Sort lists when checking for changes

When two lists are checked for changes in this module, the lists are 
reported changed when the order of the items is different. This PR 
resolves this issue.

* docker_swarm_service: Minor typo fix

* docker_swarm_service: Another minor typo

* docker_swarm_service: Should use sorted(), not sort()

* docker_swarm_service: Sort lists of dictionaries

* docker_swarm_service: Fix style issues in tests

* docker_swarm_service: Updates to integration tests

* docker_swarm_service: Casting string types within lists when comparing

* docker_swarm_service: Special handling of unordered networks with ordered aliases

* docker_swarm_service: Sorting network lists

* docker_swarm_serivce: Better unit test code coverage for lists and networks

* docker_swarm_service: Fixed coding style for sanity tests

* docker_swarm_service: More coding style fixes

* docker_swarm_service: Ignoring test for Python < 3

* docker_swarm_service: Update to version info check for backwards compatibility

* docker_swarm_service: Added change fragment #63887

* docker_swarm_service: Better handling of missing sort key for dictionary of lists

* docker_swarm_service: Preventing sorts from modifying in-place

Co-Authored-By: Felix Fontein <felix@fontein.de>

* docker_swarm_service: Removed spurious import in test

* docker_swarm_service: Preventing sorts from modifying more data in-place

Co-Authored-By: Felix Fontein <felix@fontein.de>

akamch committed:
Fix typos in `ansible.cfg` comments

PR #65177

Commit:
Merge branch 'devel' of https://github.com/netmonk/ansible into devel

Commit:
fixing merge with upstream devel

Commit:
evolving test

Commit:
fixed append arugment for command in lib, and wrote success test for latin1 dump

netmonk committed:
Adding test for import, and updating related documentation for import and dump

netmonk commented:
Can someone destroy and delete this PR... The rebase is not welcome, time to start again from fresh fork... thank you !
